luigi/rpc.py

```diff
@@ -42,32 +57,62 @@ def __init__(self, message, sub_exception=None):
         self.sub_exception = sub_exception
 
 
+class FetcherException(Exception):
+    def __init__(self, original_exc):
+        self.original_exc = original_exc
+
+
+class URLLibFetcher(object):
+    def fetch(self, full_url, body, timeout):
+        try:
+            body = urlencode(body).encode('utf-8')
+            return urlopen(full_url, body, timeout).read().decode('utf-8')
+        except (URLError, socket.timeout) as e:
+            raise FetcherException(e)
+
+
+class RequestsFetcher(object):
+    def __init__(self, session):
+        self.session = session
+
+    def fetch(self, full_url, body, timeout):
+        from requests import exceptions as requests_exceptions
+        try:
+            resp = self.session.get(full_url, data=body, timeout=timeout)
+            resp.raise_for_status()
+            return resp.text
+        except (requests_exceptions.RequestException) as e:
+            raise FetcherException(e)
+
+
 class RemoteScheduler(Scheduler):
     """
     Scheduler proxy object. Talks to a RemoteSchedulerResponder.
     """
 
-    def __init__(self, host='localhost', port=8082, connect_timeout=None, url_prefix=''):
-        self._host = host
-        self._port = port
-        self._url_prefix = url_prefix
+    def __init__(self, url='http://localhost:8082/', connect_timeout=None):
+        assert (
+            not (url.startswith('http+unix://') and not HAS_UNIX_SOCKETS),
+            'You need to install requests-unixsocket for Unix socket support.',
+        )
 
+        self._url = url
```

Comment:
If trailing `/` is present on `url`, fetcher gets lots of 404s for path `/app/...`
So it's better to do:

    self._url = url.rstrip('/')